locksmith/src/operations/checkoutConfigOperations.ts

```diff
@@ -1,42 +1,132 @@
+ */
+const extractLockInfo = (
+  config: any
+): { address: string; network: number }[] => {
+  const defaultNetwork = config.network
+  return Object.entries(config.locks).map(
+    ([address, lockConfig]: [string, any]) => ({
+      address,
+      network: lockConfig.network || defaultNetwork,
+    })
+  )
+}
+
+/**
+ * Checks if a user is a manager for a given lock
+ * @param lockAddress - The address of the lock
+ * @param userAddress - The address of the user
+ * @param network - The network ID
+ * @returns A boolean indicating whether the user is a lock manager
+ */
+const isLockManager = async (
+  lockAddress: string,
+  userAddress: string,
+  network: number
+): Promise<boolean> => {
+  const web3Service = new Web3Service(networks)
+  try {
+    return await web3Service.isLockManager(lockAddress, userAddress, network)
+  } catch (error) {
+    console.error(`Error checking lock manager status: ${error}`)
+    return false
+  }
+}
+
+/**
+ * Determines if a user is authorized to manage a checkout config
+ * @param userAddress - The address of the user
+ * @param checkoutConfig - The checkout configuration object
+ * @returns A boolean indicating whether the user is authorized
+ */
+const isUserAuthorized = async (
+  userAddress: string,
+  checkoutConfig: CheckoutConfig
+): Promise<boolean> => {
+  // first check if user is the creator
+  if (checkoutConfig.createdBy === userAddress) {
+    return true
+  }
+
+  const lockInfo = extractLockInfo(checkoutConfig.config)
+  for (const { address, network } of lockInfo) {
+    if (await isLockManager(address, userAddress, network)) {
+      return true
+    }
+  }
+
+  return false
+}
+
+/**
+ * Creates or updates a checkout configuration
+ * @param args - The SaveCheckoutConfigArgs object
+ * @returns The created or updated checkout configuration or null if not authorized
+ */
 export const saveCheckoutConfig = async ({
   id,
   name,
-  createdBy,
+  user,
   config,
-}: SaveCheckoutConfigArgs) => {
-  const generatedId = randomUUID()
+}: SaveCheckoutConfigArgs): Promise<CheckoutConfig | null> => {
+  const generatedId = id || randomUUID()
 
-  // Forcing the referrer to exist and be set to the creator of the config
-  if (!config.referrer) {
-    config.referrer = createdBy
+  const checkoutConfigData = await PaywallConfig.strip().parseAsync(config)
+
+  // Ensure referrer is set to the user
+  if (!checkoutConfigData.referrer) {
+    checkoutConfigData.referrer = user
+  }
+
+  // Check if the user is authorized to save/update the config
+  if (id) {
+    const existingConfig = await CheckoutConfig.findOne({ where: { id } })
+    if (existingConfig && !(await isUserAuthorized(user, existingConfig))) {
+      return null
+    }
   }
 
   const [createdConfig] = await CheckoutConfig.upsert(
     {
+      id: generatedId,
       name,
-      id: id || generatedId,
-      config,
-      createdBy,
+      config: checkoutConfigData,
+      createdBy: user,
```

Comment:
If this is an existing config I worry we are overwriting the `createdBy` field...

```suggestion
      createdBy: existingConfig.createdBy | user,
```

Reply:
@0xTxbi please check this